code/src/Components/ThoughtsForm.js

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { HeartIcon } from "./HeartIcon";
+
+export const ThoughtsForm = ({ onFormSubmit, setNewMessages, newMessages }) => {
+  const handleChange = (event) => {
+    setNewMessages(event.target.value);
+  };
+
+  return (
+    <form onSubmit={onFormSubmit}>
+      <h1 className="header-text">What's making you happy right now?</h1>
+      <textarea
+        className="input"
+        type="text"
+        value={newMessages}
+        onChange={handleChange}
+        placeholder="Type Happy thought here..."
+      />
+      <div className="counter-container">
+        <button
+          className="form-button"
+          disabled={newMessages.length < 6 || newMessages.length > 140}
```

Comment:
I like that you are showing a counter below the form input that updates as the user types and shows how many characters are remaining.